lib/CodeGen/CodeGenModule.cpp

```diff
@@ -1332,9 +1332,16 @@ static void addLinkOptionsPostorder(CodeGenModule &CGM, Module *Mod,
     // Link against a framework.  Frameworks are currently Darwin only, so we
     // don't to ask TargetCodeGenInfo for the spelling of the linker option.
     if (Mod->LinkLibraries[I-1].IsFramework) {
+      StringRef LibName = Mod->LinkLibraries[I - 1].Library;
+
+      // For modules that use export_as for linking, use that module
+      // name instead.
```

Comment:
It seems like we should support `export_as` with non-frameworks as well, but maybe that's not important right now.